server/src/cli/accounts.rs

```diff
@@ -1,43 +1,74 @@
+use cli_table::{Cell as _, Style as _, Table as _};
 use database::schema::Account;
 use structopt::StructOpt;
 
+#[derive(StructOpt, Debug)]
+pub struct Command {
+    /// The ID of the account
+    #[structopt(long)]
+    id: Option<i64>,
+
+    /// The Twitch handle to look up to find the account
+    #[structopt(long)]
+    twitch: Option<String>,
+
+    /// The command to execute
+    #[structopt(subcommand)]
+    operation: Operation,
+}
+
 #[derive(StructOpt, Debug)]
 #[structopt(about = "commands to execute")]
 /// commands that the server can execute
-pub enum AccountCommand {
+enum Operation {
+    /// Views the account
+    View,
     /// Sets the superuser flag on an account
     SetSuperUser,
     /// Clears the superuser flag on an account
     SetNormalUser,
 }
 
-pub async fn handle_command(
-    id: Option<i64>,
-    twitch: Option<String>,
-    command: AccountCommand,
-) -> anyhow::Result<()> {
+pub async fn handle_command(command: Command) -> anyhow::Result<()> {
     database::initialize().await;
-    let account = if let Some(id) = id {
+    let account = if let Some(id) = command.id {
         Account::load(id, database::pool()).await?
-    } else if let Some(twitch) = twitch {
-        Account::find_by_twitch_username(&twitch, database::pool()).await?
+    } else if let Some(twitch) = &command.twitch {
+        Account::find_by_twitch_username(twitch, database::pool()).await?
     } else {
         anyhow::bail!("Either id or twitch parameters must be specified for account commands")
     };
 
     match account {
-        Some(mut account) => match command {
-            AccountCommand::SetSuperUser => {
-                account.superuser = true;
-                account.save(database::pool()).await?;
-                Ok(())
+        Some(mut account) => {
+            match command.operation {
+                Operation::SetSuperUser => {
+                    account.superuser = true;
+                    account.save(database::pool()).await?;
+                }
+                Operation::SetNormalUser => {
+                    account.superuser = false;
+                    account.save(database::pool()).await?;
+                }
+                Operation::View => {}
             }
-            AccountCommand::SetNormalUser => {
-                account.superuser = false;
-                account.save(database::pool()).await?;
-                Ok(())
-            }
-        },
+
+            print_account(account)?;
+
+            Ok(())
+        }
         None => anyhow::bail!("Account not found"),
     }
 }
+
+fn print_account(account: Account) -> std::io::Result<()> {
```

Comment:
There is a derive macro, so you don't have to implement it by hand: https://docs.rs/cli-table/0.4.5/cli_table/derive.Table.html, also has all sorts of options.

This should be a method at minimum and `Display` at best, for the latter I wouldn't know what to do with the `Result` though.

Reply:
Are we not concerned about the impact of proc macros on compilation performance? ... but... in a point to you, I didn't set `default-featulres = false` so the feature was enabled already. I was originally avoiding the proc macro for that reason, even though I forgot to disable it.

Reply:
Personally I would never do the tradeoff between compile time and code quality, if a proc macro increases code quality, I won't care about compile time.